src/Fixer/Basic/Psr4Fixer.php

```diff
@@ -70,7 +70,12 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
 
                 $classyIndex = $tokens->getNextMeaningfulToken($index);
                 $classyName = $tokens[$classyIndex]->getContent();
-            }
+
+                $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
+                if ($prevToken->isGivenKind(T_NEW)) {
+                    return;
```

Comment:
can't determining that class is anonymous be done before determining it;'s name ?